test/validate/utils.go

```diff
@@ -11,14 +11,16 @@ import (
 )
 
 func compareIPs(expected map[string]string, actual []string) error {
-	if len(expected) != len(actual) {
-		return errors.Errorf("len of expected IPs != len of actual IPs, expected: %+v, actual: %+v", expected, actual)
-	}
+	expectedLen := len(expected)
 
 	for _, ip := range actual {
 		if _, ok := expected[ip]; !ok {
 			return errors.Errorf("actual ip %s is unexpected, expected: %+v, actual: %+v", ip, expected, actual)
 		}
+		delete(expected, ip)
```

Comment:
Not able to understand why `expected` is being manipulated here ?

Reply:
Manipulating expected like this allows us to get any leaked IP(s) (as it will be the only IPs left in the map after all actual IPs are deleted) if the total number of IPs dont match between actual and expected.

Per PR description - "Adds in reporting for leaked ips that exist on state files when compareIPs() fails due to size difference between actual IPs on the cluster and what exists in the state file. Old functionality reported an entire map of IPs from both current IPs and state file IPs"

Reply:
I see, can we add that to the error message that these Ips might be leaked.